docs/grouping.md

```diff
@@ -135,6 +136,77 @@ tw`
 
 In the above example, the `after` pseudo element will only be applied upon hover.
 
+
+## Thinking in Groups
+
+Take a look at this button implemented in Tailwind. It requires quite a few classes to achieve the desired functionality:
+
+```html
+  <button class="w-full sm:w-auto text-lg uppercase text-gray-100 bg-purple-800 hover:bg-purple-700 focus:bg-purple-700 focus-visible:ring-4 ring-purple-400 px-6 py-2 rounded-full transition-colors duration-300">
+    Click Me
+  </button>
+  ```
+
+Let's see how we can use grouping to improve readability and save a few characters in the process. 
+
+**The width classes** 
+
+These classes can be grouped using directive grouping:
+
+Before: `w-full sm:w-auto`
+
+After: `w(full sm:auto)`
+
+**The text classes**
+
+Twind convienently provides a `text-uppercase` class, which allows us to group all of the text-related classes together also using directive grouping:
+
+Before: `text-lg uppercase text-gray-100`
+
+After: `text(lg uppercase gray-100)`
+
+**The background classes**
+
+The background classes can be grouped in several combinations. Here are a couple options:
+
+1. Using a combination of directive and variant (mixed) grouping:
+
+    Before: `bg-purple-800 hover:bg-purple-700 focus:bg-purple-700`
+
+    After: `bg(purple-800 hover:(purple-700) focus:(purple-700))`
+
+2. Using a combination of mixed grouping and self-referencing:
+
+    Before: `bg-purple-800 hover:bg-purple-700 focus:bg-purple-700`
+
+    After: `bg(purple(800 700(hover:& focus:&)))`
```

Comment:
I think `bg-purple(800 700(hover:& focus:&))` is a _little_ bit more readable.

Reply:
I agree and it saves a character. Every bit counts!